src/Persistence_matrix/include/gudhi/base_matrix/base_matrix_1010.h

```diff
@@ -0,0 +1,454 @@
+	template<class Container_type>
+	Base_matrix_with_column_compression_with_row_access(const std::vector<Container_type>& columns);
+	Base_matrix_with_column_compression_with_row_access(unsigned int numberOfColumns);
+	Base_matrix_with_column_compression_with_row_access(const Base_matrix_with_column_compression_with_row_access& matrixToCopy);
+	Base_matrix_with_column_compression_with_row_access(Base_matrix_with_column_compression_with_row_access&& other) noexcept;
+
+	template<class Container_type>
+	void insert_column(const Container_type& column);
+	template<class Boundary_type>
+	void insert_boundary(const Boundary_type& boundary);
+	Column_type& get_column(index columnIndex);
+	Row_type& get_row(index rowIndex);
+	const Row_type& get_row(index rowIndex) const;
+	void erase_column(index columnIndex);
+	void erase_row(index rowIndex);
+
+	unsigned int get_number_of_columns() const;
+
+	void add_to(index sourceColumnIndex, index targetColumnIndex);
+	void add_to(const Column_type& sourceColumn, index targetColumnIndex);
+	void add_to(const Column_type& sourceColumn, const Field_element_type& coefficient, index targetColumnIndex);
+	void add_to(const Field_element_type& coefficient, const Column_type& sourceColumn, index targetColumnIndex);
+
+	void zero_cell(index columnIndex, index rowIndex);
+	void zero_column(index columnIndex);
+	bool is_zero_cell(index columnIndex, index rowIndex);
+	bool is_zero_column(index columnIndex);
+
+	Base_matrix_with_column_compression_with_row_access& operator=(const Base_matrix_with_column_compression_with_row_access& other);
+	friend void swap(Base_matrix_with_column_compression_with_row_access& matrix1, Base_matrix_with_column_compression_with_row_access& matrix2){
+		matrix1.rows_.swap(matrix2.rows_);
+		matrix1.columnToRep_.swap(matrix2.columnToRep_);
+		swap(matrix1.columnClasses_, matrix2.columnClasses_);
+		matrix1.repToColumn_.swap(matrix2.repToColumn_);
+		std::swap(matrix1.nextColumnIndex_, matrix2.nextColumnIndex_);
+		for (auto& p : matrix1.repToColumn_){
+			Column_type& col = p.second;
+			col.set_rows(&matrix1.rows_);
+		}
+		for (auto& p : matrix2.repToColumn_){
+			Column_type& col = p.second;
+			col.set_rows(&matrix2.rows_);
+		}
+	}
+
+	void print();  //for debug
+
+private:
+	using rows_type = typename Master_matrix::row_container_type;
+	using cell_rep_type = typename std::conditional<
+								Master_matrix::Option_list::is_z2,
+								index,
+								std::pair<index,typename Master_matrix::Field_type>
+							>::type;
+	using tmp_column_type = typename std::conditional<
+								Master_matrix::Option_list::is_z2,
+								std::set<index>,
+								std::set<std::pair<index,Field_element_type>,CellPairComparator<Field_element_type> >
+							>::type;
+
+	struct ColumnPointerHash {
+		size_t operator()(const Column_type* c) const
+		{
+			return std::hash<Column_type>()(*c);
+		}
+	};
+
+	struct ColumnPointerEqual : std::binary_function<const Column_type*, const Column_type*, bool> {
```

Comment:
`std::binary_function` was deprecated in C++11 and removed in C++17 (what we currently use).